build_test.go

```diff
@@ -8,13 +8,13 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
-func TestCommand(t *testing.T) {
+// for yum packages, test the user has an id of 0 (root)
+func TestUserCommand(t *testing.T) {
```

Comment:
Is this really a _test_, though? I mean, test functions in Go should test some behaviour of our _code_, not something about the user. I'm guessing that this should really be in the `thing` package and it wants to verify, before trying to run a `yum` command, for example, that the user actually has the right privileges to do that.

But maybe we don't really need to explicitly verify that. If the user doesn't have root powers, then the `yum` command will presumably fail with an error, so we'll catch that already in `r.Command()` and return it.

You could imagine a function called `r.MustBeRoot()`, which you can call from `main` and it terminates the program if the user is not `root`:

```go
func main() {
	r := thing.NewRunner()
	r.MustBeRoot() // terminates if user is not root
	r.DoStuff(...)
}
```

Reply:
that's exactly what I was thinking about because yum does require sudo privileges. 

I guess we only need to explain it in the readme file that the user will need to run the package via sudo for yum but not for gem, since that is installed in user space. 

if this is deployed as one binary...so I will need to think about how to separate those executions so gems are not installed as root.